dttools/src/disk_alloc.c

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright (C) 2015- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/mount.h>
+#include <string.h>
+#include <ctype.h>
+#include <errno.h>
+
+#include "disk_alloc.h"
+
+int disk_alloc_create(char *loc, char *space) {
+
+	long long size;
+	char metric;
+	int result;
+
+	//Check for trailing '/'
+	if(loc[strlen(loc) - 1] == 47) {
+
+		loc[strlen(loc) - 1] = 0;
+	}
+
+	//Scale disk allocation to correct size
+	if(sscanf(space, "%lli%c", &size, &metric) == 2) {
+
+		switch (toupper((int) metric)) {
+
+			case 'P':
+				size = size << 10;
+			case 'T':
+				size = size << 10;
+			case 'G':
+				size = size << 10;
+			case 'M':
+				size = size << 10;
+			case 'K':
+				break;
+		}
+
+	}
+
+	else {
+
+		printf("\nInvalid device size parameters passed.\n");
```

Comment:
A library function should (generally) not print to stdout -- that's the job of main().
Instead, return failure and set errno=EINVAL.